### PR TITLE
Improve guidance for IPC transports, discuss using multiple

### PR DESCRIPTION
**Pull request by JamesNK**
[Internal review](https://review.learn.microsoft.com/en-us/aspnet/core/grpc/interprocess?view=aspnetcore-8.0&branch=pr-en-us-28282)

### CODE REVIEW

File: aspnetcore/grpc/interprocess.md [hunk 1]
```diff
@@ -42,10 +42,33 @@ gRPC calls between a client and server on different machines are usually sent ov
...
+
+Cross-platform apps may want to use different IPC transports, depending on the current OS. An app can check the current operating system on startup and chose the desired transport for that platform:
+
+```csharp
+var builder = WebApplication.CreateBuilder(args);
+builder.WebHost.ConfigureKestrel(serverOptions =>
+{
+    if (OperatingSystem.IsWindows())
+    {
+        serverOptions.ListenNamedPipe("MyPipeName", listenOptions =>
```
Comment (gfoidl) [inline aspnetcore/grpc/interprocess.md:57]: For named pipes (above) there's a hint about Windows security, but can you add a note about perf (like done for UDS) too?

So I head the question coming: "When I don't need Win-security, why should I use named pipes on Windows when I could use UDS too?"

File: aspnetcore/grpc/interprocess.md [hunk 2]
```diff
@@ -42,10 +42,33 @@ gRPC calls between a client and server on different machines are usually sent ov
 
 There are transports optimized for IPC workloads. .NET supports multiple IPC transports:
 
-* [Unix domain sockets (UDS)](https://wikipedia.org/wiki/Unix_domain_socket) is a widely supported IPC technology that's more efficient than TCP when the client and server are on the same machine. UDS is usable on Linux, macOS, and [Windows 10/Windows Server 2019 or later](https://devblogs.microsoft.com/commandline/af_unix-comes-to-windows/).
-* [Named pipes](https://wikipedia.org/wiki/Named_pipe) are supported by all versions of Windows. Named pipes integrate well with [Windows security](/windows/win32/ipc/named-pipe-security-and-access-rights) to control client access to the pipe.
-
-Implement other IPC technologies using the extensibility in Kestrel and `SocketsHttpHandler`.
+* [Unix domain sockets (UDS)](https://wikipedia.org/wiki/Unix_domain_socket) is a widely supported IPC technology that's more efficient than TCP when the client and server are on the same machine. UDS is the best choice for building cross-platform apps, and it is usable on Linux, macOS, and [Windows 10/Windows Server 2019 or later](https://devblogs.microsoft.com/commandline/af_unix-comes-to-windows/).
```
Comment (wadepickett) [inline aspnetcore/grpc/interprocess.md:45]:
```suggestion
* [Unix domain sockets (UDS)](https://wikipedia.org/wiki/Unix_domain_socket) is a widely supported IPC technology that's more efficient than TCP when the client and server are on the same machine. UDS is the best choice for building cross-platform apps, and it's usable on Linux, macOS, and [Windows 10/Windows Server 2019 or later](https://devblogs.microsoft.com/commandline/af_unix-comes-to-windows/).
```

File: aspnetcore/grpc/interprocess.md [hunk 3]
```diff
@@ -42,10 +42,33 @@ gRPC calls between a client and server on different machines are usually sent ov
...
+* [Named pipes](https://wikipedia.org/wiki/Named_pipe) are supported by all versions of Windows. Named pipes integrate well with [Windows security](/windows/win32/ipc/named-pipe-security-and-access-rights), which can be used to control client access to the pipe.
+* Other IPC transports by implementing <xref:Microsoft.AspNetCore.Connections.IConnectionListenerFactory> and register the implementation at app startup.
```
Comment (wadepickett) [inline aspnetcore/grpc/interprocess.md:47]:
```suggestion
* Other IPC transports by implementing <xref:Microsoft.AspNetCore.Connections.IConnectionListenerFactory> and registering the implementation at app startup.
```

**Review by wadepickett — APPROVED** [review]
Looks good to me.  Just some very tiny nits.  
Consider updating the date in the metadata to show the doc is up to date.

**Pull request merged**